Meadowlark-js/packages/meadowlark-core/src/model/DocumentId.ts

```diff
@@ -44,16 +45,25 @@ export function documentIdForDocumentIdentity(
 }
 
 /**
- * Returns the id of the given DocumentInfo, using the project name, resource name, resource version
- * and identity of the API document.
- * If the given documentInfo is assignable, uses that information for the document id.
+ * Returns the id of the given DocumentInfo, using the project name, resource name and identity of the API document.
  */
 export function documentIdForDocumentInfo(resourceInfo: ResourceInfo, documentInfo: DocumentInfo): string {
-  // If this is an assignable entity, use the assignableIdentity for the id instead of the actual document identity
-  const documentIdentity: DocumentIdentity =
-    documentInfo.assignableInfo == null ? documentInfo.documentIdentity : documentInfo.assignableInfo.assignableIdentity;
```

Comment:
This was convoluted, but removing broke Dynamo's superclass handling.

Reply:
So... we end up with a function that just delegates to another function? Can we cut out the middleman? Or would it make sense to move this function into `DocumentInfo`?

Reply:
👍